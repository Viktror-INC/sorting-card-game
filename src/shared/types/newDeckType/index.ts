export interface INewDeck {
  deck_id: string;
  remaining: number;
  shuffled: boolean;
  success: boolean;
}

export interface IMyCards {
  code: string;
  image: string;
  images: {
    svg: string;
    png: string;
  };
  value: string;
  suit: string;
  showed: boolean;
}
