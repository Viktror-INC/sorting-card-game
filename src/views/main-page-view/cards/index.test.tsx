import {
  render,
  screen,
  fireEvent,
  cleanup,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import Cards from ".";
import { IMyCards } from "@/shared/types/newDeckType";

const myCards = [
  {
    code: "3D",
    image: "https://deckofcardsapi.com/static/img/3D.png",
    images: {
      svg: "https://deckofcardsapi.com/static/img/3D.svg",
      png: "https://deckofcardsapi.com/static/img/3D.png",
    },
    value: "3",
    suit: "DIAMONDS",
    showed: true,
  },
  {
    code: "6S",
    image: "https://deckofcardsapi.com/static/img/6S.png",
    images: {
      svg: "https://deckofcardsapi.com/static/img/6S.svg",
      png: "https://deckofcardsapi.com/static/img/6S.png",
    },
    value: "6",
    suit: "SPADES",
    showed: true,
  },
  {
    code: "7S",
    image: "https://deckofcardsapi.com/static/img/7S.png",
    images: {
      svg: "https://deckofcardsapi.com/static/img/7S.svg",
      png: "https://deckofcardsapi.com/static/img/7S.png",
    },
    value: "7",
    suit: "SPADES",
    showed: true,
  },
  {
    code: "9S",
    image: "https://deckofcardsapi.com/static/img/9S.png",
    images: {
      svg: "https://deckofcardsapi.com/static/img/9S.svg",
      png: "https://deckofcardsapi.com/static/img/9S.png",
    },
    value: "9",
    suit: "SPADES",
    showed: true,
  },
  {
    code: "0D",
    image: "https://deckofcardsapi.com/static/img/0D.png",
    images: {
      svg: "https://deckofcardsapi.com/static/img/0D.svg",
      png: "https://deckofcardsapi.com/static/img/0D.png",
    },
    value: "10",
    suit: "DIAMONDS",
    showed: true,
  },
];

const onCardClick = jest.fn();

describe("cards test", () => {
  const renderCards = (cards?: IMyCards[]) =>
    render(<Cards cards={cards ? cards : myCards} onCardClick={onCardClick} />);

  afterEach(() => {
    cleanup();
  });

  it("correct value", () => {
    const renderedCards = renderCards();
    const cards = screen.getAllByTestId(/card-/);
    expect(cards).toHaveLength(5);
    cards.forEach((_, index) => {
      const card = screen.getByTestId(`card-${index}`);
      expect(card).toBeInTheDocument;
      fireEvent.click(card);
      expect(onCardClick).toBeCalled();
    });

    const cardsNotFound = screen.queryByTestId("cards-not-found");
    expect(cardsNotFound).not.toBeInTheDocument();

    expect(renderedCards).toMatchSnapshot();
  });

  it("empty array", () => {
    renderCards([]);
    const cards = screen.queryAllByTestId(/card-/);
    expect(cards).toHaveLength(0);
    const cardsNotFound = screen.getByTestId("cards-not-found");
    expect(cardsNotFound).toBeInTheDocument();
    expect(cardsNotFound).toContainHTML("Cards not found");
  });
});
