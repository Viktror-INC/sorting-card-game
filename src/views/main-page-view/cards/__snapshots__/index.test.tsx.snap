// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cards test correct value 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="flex align-middle items-center justify-center gap-x-5"
      >
        <li
          class="flex"
          data-testid="card-0"
        >
          <img
            alt="3D"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F3D.png&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F3D.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F3D.png&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
        </li>
        <li
          class="flex"
          data-testid="card-1"
        >
          <img
            alt="6S"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F6S.png&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F6S.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F6S.png&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
        </li>
        <li
          class="flex"
          data-testid="card-2"
        >
          <img
            alt="7S"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F7S.png&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F7S.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F7S.png&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
        </li>
        <li
          class="flex"
          data-testid="card-3"
        >
          <img
            alt="9S"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F9S.png&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F9S.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F9S.png&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
        </li>
        <li
          class="flex"
          data-testid="card-4"
        >
          <img
            alt="0D"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F0D.png&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F0D.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F0D.png&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flex align-middle items-center justify-center gap-x-5"
    >
      <li
        class="flex"
        data-testid="card-0"
      >
        <img
          alt="3D"
          data-nimg="1"
          decoding="async"
          height="400"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F3D.png&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F3D.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F3D.png&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </li>
      <li
        class="flex"
        data-testid="card-1"
      >
        <img
          alt="6S"
          data-nimg="1"
          decoding="async"
          height="400"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F6S.png&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F6S.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F6S.png&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </li>
      <li
        class="flex"
        data-testid="card-2"
      >
        <img
          alt="7S"
          data-nimg="1"
          decoding="async"
          height="400"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F7S.png&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F7S.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F7S.png&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </li>
      <li
        class="flex"
        data-testid="card-3"
      >
        <img
          alt="9S"
          data-nimg="1"
          decoding="async"
          height="400"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F9S.png&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F9S.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F9S.png&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </li>
      <li
        class="flex"
        data-testid="card-4"
      >
        <img
          alt="0D"
          data-nimg="1"
          decoding="async"
          height="400"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F0D.png&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F0D.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdeckofcardsapi.com%2Fstatic%2Fimg%2F0D.png&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
